barman/cli.py

```diff
@@ -1810,6 +1814,32 @@ def check_wal_archive(args):
             output.close_and_exit()
 
 
+@command(
+    [
+        argument(
+            "server_name",
```

Comment:
It definitely feels weird specifying the `server_name` here but then also specifying the server name as a key in `json_changes` and my instinct is to remove this parameter altogether and just rely on the keys supplied in the json - if they map to a section in the config file then changes are applied there, otherwise it's an error.

This would allow things like `["barman": {...}}` to update the global config, which would help Barman play nicely with external automation beyond the current HA scope.

Reply:
I was thinking about alternative formats for `json_data` which might make bulk changes easier - something like:

```
[
  {
    "scope": "global",
    "log_level": "DEBUG"
  },
  {
    "scope": "server",
    "server_name": "main",
    "conninfo": "db=main.local"
  },
  {
    "scope": "model",
    "server_name": "main",
    "model_name": "main-failover",
    "conninfo": "db=failover.local"
  },
  {
    "scope": "model",
    "server_name": "main",
    "model_name": "all",
    "primary_conninfo": "main-failover"
  },
]
```

The proposal being:
- The JSON body is a list where each item is a set of config changes.
- `scope`, `server_name` and `model_name` are special keys which define the scope (global/server/model) and the server and model (if required by the scope).
- `model_name` supports a special value `all` which causes the change to be applied to all models for a server.

This would need further refinement but as a general idea does it make (any) sense? It's a step away from mapping input JSON directly to sections but would allow more flexibility in how external actors specify changes.

Reply:
I do like that idea. The only thing that sounded weird to me at a first glance is the models part:

```
  {
    "scope": "model",
    "server_name": "main",
    "model_name": "main-failover",
    "conninfo": "db=failover.local"
  },
  ```
  
  I think we should not "tie" models with servers given both servers and models belong to clusters. IMHO we should use the 3 scopes (either global, server or scope) and do not mix `server_name` and `model_name`. Thoughts?

Reply:
Some side thought too: We will likely "get rid" of this command before merging, right? I mean, there are some separate issues in the board to handle the commands -- unless we decide referencing those through this PR and shipping all together.

Reply:
yeah i do agree with this and is something I was thinking myself... I need to check how to correctly handle the server loading without putting much weight in the cli module. cycling on this atm

Reply:
i'm not sure we want this command to be able to write global/barman-section configurations...
besides this... i was thinking around the scope etc... well if we put aside global for a moment... we have the `server` and `model` scope.. but in the end a `model` is a "normal" config + `model = true` so my question is... does it need a scope? @mikewallace1979 @barthisrael

Reply:
keeping the scope around for now... what do you think of this format:
```
{
    "main": { 
        "scope": "server", 
        "conninfo": "db=main.local" }, 
    "main": {
        "scope": "model",
        "model_name": "main-failover", 
        "conninfo": "db=failover.local" }, 
    "main": { 
        "scope": "model", "model_name": "all", 
        "primary_conninfo": "main-failover" 
    }
}
```

In terms of readability and order using the servername as identifier sorta makes sense...
OTOH, the above example seems exceptionally weird as were have 3 "main" labels (and I'm not even sure is ok in terms of json)... open to discussions @barthisrael @mikewallace1979

Reply:
I wonder if we should use a list of dicts, instead of a dict of dicts for the payload.
I mean, something like this:

```
[
    {
        "scope": "server",
        "server_name": "main",
        "conninfo": "db=main.local"
    }, 
    {
        "scope": "model",
        "model_name": "main-failover", 
        "conninfo": "db=failover.local"
    }, 
    { 
        "scope": "model",
        "model_name": "all", 
        "primary_conninfo": "main-failover" 
    }
]
```

I mean, it sounds more aligned with what we want to do. Also, my understanding is that models are tied to clusters, and servers are tied to clusters, but models are not tied to servers. So I guess the model specification in your example would sound a bit weird in that case.

What do you think?